lib/framework/data_context.dart

```diff
@@ -569,74 +569,41 @@ class EnsembleUtils with Invokable {
   List<Map<String, dynamic>> getCountry(String val) {
     String input = val.toLowerCase().trim();
     List<Map<String, dynamic>> alpha2List = [];
-    for (var i in allCountries) {
-      String countryCode = i['iso']['alpha-2'];
-      countryCode = countryCode.toLowerCase();
-      if (countryCode == input) {
-        alpha2List.add(i);
+    if (input.length == 2) {
+      for (var i in allCountries) {
+        String countryCode = i['iso']['alpha-2'];
+        countryCode = countryCode.toLowerCase();
+        if (countryCode == input) {
+          alpha2List.add(i);
+        }
```

Comment:
The value is unique. You need to break out and return once found. There is no reason to loop through the entire list.
The return type should be `Map<String, dynamic>?`. Either you find the single result or you don't.